crates/utils/src/response.rs

```diff
@@ -11,19 +11,20 @@ pub fn jsonify_plain_text_errors<BODY>(
     return Ok(ErrorHandlerResponse::Response(res.map_into_left_body()));
   }
   // We're assuming that any LemmyError is already in JSON format, so we don't need to do anything
-  if maybe_error
-    .expect("http responses with 400-599 statuses should have an error object")
-    .as_error::<LemmyError>()
-    .is_some()
-  {
-    return Ok(ErrorHandlerResponse::Response(res.map_into_left_body()));
+  if let Some(maybe_error) = maybe_error {
```

Comment:
I'm not sure about my changes to this function, because its a bit messy, and the early returns are confusing me as to the flow.